tests/unit/src/main/scala/tests/TestingClient.scala

```diff
@@ -349,7 +348,18 @@ class TestingClient(workspace: AbsolutePath, val buffers: Buffers)
       params: PublishDecorationsParams
   ): Unit = {
     val path = params.uri.toAbsolutePath
-    decorations.put(path, params.options)
+    decorations.compute(
+      path,
+      {
+        case (_, decorationTypes) => {
+          if (decorationTypes == null) {
+            Set(params)
+          } else {
+            decorationTypes.filter(p => p.isInline != params.isInline) + params
```

Comment:
Why are you filtering decorations?

Reply:
To replace the particular inline type. If `isInline=true` we will only replace decorations with inline=true and vice versa